ocaml/gencert/pem.ml

```diff
@@ -38,9 +38,11 @@ let ws = take_while is_whitespace
 
 let data = take_while1 is_data
 
-let key_header = string "-----BEGIN PRIVATE KEY-----"
+let kind = string " RSA " <|> string " EC " <|> string " "
 
-let key_footer = string "-----END PRIVATE KEY-----"
+let key_header = string "-----BEGIN" <* kind <* string "PRIVATE KEY-----"
+
+let key_footer = string "-----END" <* kind <* string "PRIVATE KEY-----"
```

Comment:
This doesn't ensure that the footer matches the header, is this something that should worry us?

It doesn't ensure that the footer and the header matches the content, which is probably something that is more important than the previous issue. If he kind is kept as a variant integrity between all three components can be ensured.

Reply:
I agree that this is not checked - however, these are just delimiters around the key, which is checked for integrity at later stages. I could read the kind of the header and then expect it again at the footer.

Reply:
I fixed that (and another problem) - so we should be more precise in what we recognize.